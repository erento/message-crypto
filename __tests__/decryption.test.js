const messageCrypto = require('../index.js');

describe('decryption', () => {
    test('should decrypt a message', () => {
        return expect(messageCrypto.decrypt('wi/WnD0ujMRoTG/d6cAGufSZ0rxaNE8uUXFE+eYqfR4=', 'TestAESKey'))
            .resolves
            .toMatchSnapshot();
    });

    test('should fail to decrypt a message', () => {
        return expect(messageCrypto.decrypt('Wi/WnD0ujMRoTG/d6cAGufSZ0rxaNE8uUXFE+eYqfR4=', 'TestAESKey'))
            .resolves
            .toMatchSnapshot();
    });

    test('should fail to decrypt a message', () => {
        return expect(messageCrypto.decrypt('wi/WnD0ujMRoTG/d6cAGufSZ0rxaNE8uUXFE+eYqfR4=', 'TestKey'))
            .rejects
            .toThrowError('error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt');
    });
});
