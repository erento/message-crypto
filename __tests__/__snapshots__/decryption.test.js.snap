// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decryption should decrypt a message 1`] = `"A123"`;

exports[`decryption should fail to decrypt a message 1`] = `"ï¿½123"`;
